ebib.el

```diff
@@ -4157,11 +4267,35 @@ all selected keys are returned as a list of strings."
   (let* ((ebib-citation-insert-multiple t)
 	 (entry (ebib-read-entry
 		 (format "%s: " prompt) databases
-		 multiple)))
+		 multiple filter)))
     (if multiple
 	(mapcar #'car entry)
       (caar entry))))
 
+(defcustom ebib-always-prompt-for-special-keys t
+  "Prompt for new key in already-named Xdata, Set entries."
+  :group 'ebib
+  :type 'boolean)
+
+(defun ebib--edit-type-field ()
+  "Prompt for an entry type.
+If the selected type is \"xdata\" or \"set\" (or a
+case-insensitive variation) and
+`ebib-always-prompt-for-special-keys' is non-nil or the original
+key is temporary or then also invoke `ebib-edit-keyname', to
+force the user to select a key for the entry manually."
+  (let ((type (completing-read
+	       "Type: "
+	       (ebib--list-entry-types (ebib--get-dialect ebib--cur-db) t)
+	       nil t)))
+    (when (and (member-ignore-case type '("xdata" "set"))
+	       (or ebib-always-prompt-for-special-keys
+		   (string-match "<new-entry[[:digit:]]+>"
```

Comment:
I get an error here because `string-match` is called with four arguments, but it only accepts 2-3. I'm running Emacs 28, but I believe it's still the same in Emacs 29.

Reply:
Thanks! Will try to fix this today.

Reply:
This should be sorted now. In 29.0.60 it has an extra arg which has the same effect as wrapping the whole thing in `save-match-data`, but I can just do that explicitly instead.